collector/filesystem_common.go

```diff
@@ -126,7 +127,15 @@ func (c *filesystemCollector) Update(ch chan<- prometheus.Metric) (err error) {
 	if err != nil {
 		return err
 	}
+	// Make sure we expose a metric once, even if there are multiple mounts
+	updated := map[string]bool{}
 	for _, s := range stats {
+		k := fmt.Sprintf("%q", s)
```

Comment:
> as structs can only be used as keys if their fields are basic types

Ah! Good to know, using the struct as key was what I tried first. Will convert it.